Zend/zend_vm_def.h

```diff
@@ -5730,12 +5730,6 @@ ZEND_VM_COLD_CONST_HANDLER(51, ZEND_CAST, CONST|TMP|VAR|CV, ANY, TYPE)
 	expr = GET_OP1_ZVAL_PTR(BP_VAR_R);
 
 	switch (opline->extended_value) {
```

Comment:
Please add an assert along the lines of `ZEND_ASSERT(opline->extended_value != _IS_BOOL && "Must use ZEND_BOOL instead");` here.

Reply:
The `ZEND_ASSERT(opline->extended_value == IS_OBJECT);` I added would also fail for that same assertion, just later in the program execution. There weren't any early returns I saw.

Added anyway to `default:`